src/compression/lz4.rs

```diff
@@ -0,0 +1,502 @@
+use std::mem;
```

Comment:
Yeah, it doesn't feel appropriate to have all this LZ4 code inside a crate for Kafka, and I think it could significantly increase the burden on maintainers and future contributors who may not have the expertise to mess with the code for an LZ4 implementation. If a library doesn't exist that suits the needs of `kafka-rust`, then I think it's more appropriate to wait for one that does than to (please excuse the bluntness, because I really do appreciate your efforts!) have a bunch of hand-written, scantily-tested, -documented, and -commented code to handle a compatibility layer.

To be honest, in the absence of a quality crate that fits our needs, I'd rather just use a crate that uses the right frame checksum and notify users that it will only work for new versions of Kafka.

@xitep what are your thoughts?